core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -158,9 +157,12 @@ public void sendMessage(RespT message) {
       InputStream resp = method.streamResponse(message);
       stream.writeMessage(resp);
       stream.flush();
+    } catch (RuntimeException e) {
+      close(Status.fromThrowable(e), new Metadata());
+      throw e;
     } catch (Throwable t) {
       close(Status.fromThrowable(t), new Metadata());
-      throw Throwables.propagate(t);
+      throw new RuntimeException(t);
```

Comment:
Seems a little dangerous to possibly transform `Error` into `RuntimeException` -- or, at the least, a bit surprising.